src/renderer/screens/swap/Form/To.js

```diff
@@ -0,0 +1,235 @@
+// @flow
+
+import Box from "~/renderer/components/Box";
+import Label from "~/renderer/components/Label";
+import { Trans } from "react-i18next";
+import { SelectAccount } from "~/renderer/components/PerCurrencySelectAccount";
+import InputCurrency from "~/renderer/components/InputCurrency";
+import React, { useCallback } from "react";
+import type {
+  AccountLike,
+  Account,
+  CryptoCurrency,
+  TokenCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import { BigNumber } from "bignumber.js";
+import styled from "styled-components";
+import SelectCurrency from "~/renderer/components/SelectCurrency";
+import Input from "~/renderer/components/Input";
+import Text from "~/renderer/components/Text";
+import ToolTip from "~/renderer/components/ToolTip";
+import Price from "~/renderer/components/Price";
+import IconPlusSmall from "~/renderer/icons/PlusSmall";
+import { openModal } from "~/renderer/actions/modals";
+import { useDispatch } from "react-redux";
+import { CurrencyOptionRow } from "~/renderer/screens/swap/Form";
+import type { CurrenciesStatus } from "@ledgerhq/live-common/lib/swap/logic";
+import InfoCircle from "~/renderer/icons/InfoCircle";
+import IconLock from "~/renderer/icons/Lock";
+import useTheme from "~/renderer/hooks/useTheme";
+import { useCurrencyAccountSelect } from "~/renderer/components/PerCurrencySelectAccount/state";
+
+const InputRight = styled(Box).attrs(() => ({
+  ff: "Inter|Medium",
+  color: "palette.text.shade60",
+  fontSize: 4,
+  justifyContent: "center",
+}))`
+  padding-right: 10px;
+`;
+
+const AddAccount = styled.div`
+  display: flex;
+  cursor: pointer;
+  &:hover {
+    text-decoration: underline;
+  }
+  align-items: center;
+  border: 1px solid ${p => p.theme.colors.palette.divider};
+  height: 48px;
+  padding: 0 15px;
+  border-radius: 4px;
+  color: ${p => p.theme.colors.palette.primary.main};
+`;
+
+const SwapInputGroup = ({
+  currencies,
+  currency: defaultCurrency,
+  fromCurrency,
+  account: defaultAccount,
+  currenciesStatus,
+  validAccounts,
+  amount,
+  rate,
+  error,
+  onAccountChange,
+  onCurrencyChange,
+}: {
+  currencies: (CryptoCurrency | TokenCurrency)[],
+  currency: ?(CryptoCurrency | TokenCurrency),
+  fromCurrency: ?(CryptoCurrency | TokenCurrency),
+  account: ?Account,
+  currenciesStatus: CurrenciesStatus,
+  validAccounts: Account[],
+  amount: ?BigNumber,
+  rate?: BigNumber,
+  error?: Error,
+  onAccountChange: (AccountLike, ?Account) => void,
+  onCurrencyChange: (CryptoCurrency | TokenCurrency) => void,
+}) => {
+  const {
+    availableAccounts,
+    currency,
+    account,
+    subAccount,
+    setCurrency,
+    setAccount,
+  } = useCurrencyAccountSelect({
+    allCurrencies: currencies,
+    allAccounts: validAccounts,
+    defaultCurrency,
+    defaultAccount,
+    fallbackToFirst: false,
+  });
+
+  const unit = currency && currency.units[0];
+  const renderOptionOverride = ({ data: currency }: any) => {
```

Comment:
maybe `useCallback` here ?

Reply:
You were actually crucial in finding a re render loop bug thingy, so thank you